agreement_contract/models/agreement.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2019 Ecosoft Co., Ltd (https://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class Agreement(models.Model):
+    _inherit = 'agreement'
+
+    contract_type = fields.Selection(
+        string="Contract Type",
+        selection=[
+            ('sale', 'Customer'),
+            ('purchase', 'Supplier'),
+        ],
+        default="sale",
+        required=True,
+    )
+    contract_count = fields.Integer(
+        string="Contract Count",
+        compute="_compute_contract_count",
+    )
+
+    @api.multi
+    def _compute_contract_count(self):
+        Contract = self.env['account.analytic.account']
```

Comment:
```suggestion
        contract = self.env['account.analytic.account']
```
![Selection_025](https://user-images.githubusercontent.com/24691983/64513477-d781f080-d312-11e9-9c9e-105c106ec021.png)